### PR TITLE
[release-1.6] Try to improve the error message in one of the possible "manifest mismatch" code paths

### LINKED ISSUE #2736: Pkg should not hard crash like this
The following error occurs:
```
PS E:\00WIP\FinEtoolsFlexStructures.jl> julia
  Activating environment at `E:\00WIP\FinEtoolsFlexStructures.jl\Project.toml`
ERROR: LoadError: TypeError: in typeassert, expected VersionNumber, got a value of type Pkg.Types.VersionSpecStacktrace:
 [1] load_urls(ctx::Pkg.Types.Context, pkgs::Vector{Pkg.Types.PackageSpec})
   @ Pkg.Operations C:\buildbot\worker\package_win64\build\usr\share\julia\stdlib\v1.6\Pkg\src\Operations.jl:545
 [2] #download_source#57
   @ C:\buildbot\worker\package_win64\build\usr\share\julia\stdlib\v1.6\Pkg\src\Operations.jl:732 [inlined]  
 [3] download_source
   @ C:\buildbot\worker\package_win64\build\usr\share\julia\stdlib\v1.6\Pkg\src\Operations.jl:731 [inlined]  
 [4] instantiate(ctx::Pkg.Types.Context; manifest::Nothing, update_registry::Bool, verbose::Bool, platform::Base.BinaryPlatforms.Platform, allow_build::Bool, allow_autoprecomp::Bool, kwargs::Base.Iterators.Pairs{Union{}, Union{}, Tuple{}, NamedTuple{(), Tuple{}}})
   @ Pkg.API C:\buildbot\worker\package_win64\build\usr\share\julia\stdlib\v1.6\Pkg\src\API.jl:1406
 [5] instantiate
   @ C:\buildbot\worker\package_win64\build\usr\share\julia\stdlib\v1.6\Pkg\src\API.jl:1325 [inlined]        
 [6] #instantiate#252
   @ C:\buildbot\worker\package_win64\build\usr\share\julia\stdlib\v1.6\Pkg\src\API.jl:1321 [inlined]        
 [7] instantiate()
   @ Pkg.API C:\buildbot\worker\package_win64\build\usr\share\julia\stdlib\v1.6\Pkg\src\API.jl:1321
 [8] top-level scope
   @ C:\Users\PetrKrysl\Portable_Julia_VSCode\assets\.julia-1.6.2-depot\config\startup.jl:6
in expression starting at C:\Users\PetrKrysl\Portable_Julia_VSCode\assets\.julia-1.6.2-depot\config\startup.jl:4
PS E:\00WIP\FinEtoolsFlexStructures.jl>
```
I get that the project and the julia versions are mismatched, but a hard crash is a problem.
For instance, when that happens in vscode, the error message is lost and the user is left wondering
what went wrong.

### PR DESCRIPTION
**Pull request by DilumAluthge**
Fixes #2736

Please note that the base (target) branch of this pull request is the `release-1.6` branch.

### CODE REVIEW

**Comment by DilumAluthge** [comment]
I'm not sure if this PR should be made directly to `release-1.6`, or if it should be made to `master` and then backported to `release-1.6`.

File: src/Operations.jl [hunk 1]
```diff
@@ -542,6 +542,7 @@ function load_urls(ctx::Context, pkgs::Vector{PackageSpec})
     urls = Dict{UUID,Vector{String}}()
     for pkg in pkgs
         uuid = pkg.uuid
+        pkg.version isa VersionNumber || throw(ErrorException("The manifest file you are using was generated by a different version of Julia and is not compatible with this Julia version"))
```
Comment (IanButterworth) [inline src/Operations.jl:545]:
```suggestion
        pkg.version isa VersionNumber || throw(ErrorException("The manifest file you are using was most likely generated by a different version of Julia and is not compatible with this Julia version"))
```

**Comment by DilumAluthge** [comment]
Is this good to merge?

**Comment by DilumAluthge** [comment]
Bump @IanButterworth do you know if this PR should be made to release-1.6 or to master?

**Comment by IanButterworth** [comment]
The function this is in doesn't exist on master anymore, so I think `release-1.6` is correct

**Pull request merged**